### PR TITLE
[Toolchain] Add success and failure callback to job request

### PR DESCRIPTION
**Pull request by jyoungyun**
This commit adds success and failure callback to job request.

ONE-vscode-DCO-1.0-Signed-off-by: Jiyoung Yun <jy910.yun@samsung.com>

### CODE REVIEW

**Comment by jyoungyun** [comment]
This pr conflicts with #701. #701 should be merged first.

**Comment by jyoungyun** [comment]
PTAL /cc @Samsung/one-vscode

File: src/Project/Job.ts [hunk 1]
```diff
@@ -16,15 +16,19 @@
 
 import {ToolArgs} from './ToolArgs';
 
+export type JobCallback = {
```
Comment (YongseopKim) [inline src/Project/Job.ts:19]: When I wrorte this code, I didn't consider JobCallback's usage enough. In the future, this type could be different from this.

Comment (YongseopKim) [reply src/Project/Job.ts:19]: And extracting it as this type looks very good.

**Review by YongseopKim — APPROVED** [review]
LGTM

**Review by llFreetimell — APPROVED** [review]
LGTM

**Pull request merged**